src/test/java/io/castle/client/CastleMergeHttpTest.java

```diff
@@ -0,0 +1,77 @@
+package io.castle.client;
+
+import io.castle.client.model.AuthenticateAction;
+import io.castle.client.model.AuthenticateFailoverStrategy;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.Assert;
+import org.junit.Test;
+import org.springframework.mock.web.MockHttpServletRequest;
+
+import javax.servlet.http.HttpServletRequest;
+
+public class CastleMergeHttpTest extends AbstractCastleHttpLayerTest {
+
+    public CastleMergeHttpTest() {
+        super(new AuthenticateFailoverStrategy(AuthenticateAction.CHALLENGE));
+    }
+
+    @Test
+    public void mergeContextIsSendOnHttp() throws InterruptedException {
+
+        //Given
+        server.enqueue(new MockResponse().setResponseCode(200));
+        String id = "12345";
+
+        // And a mock Request
+        HttpServletRequest request = new MockHttpServletRequest();
+        // And a extra context
+        CustomExtraContext customExtraContext = new CustomExtraContext();
+        customExtraContext.setAddString("String value");
+        customExtraContext.setCondition(true);
+        customExtraContext.setValue(10L);
+        // and trait object
+        CustomAppTraits trait = new CustomAppTraits();
+        trait.setX("valueX");
+        trait.setY(234567);
+
+        // and an authenticate request is made
+        sdk.onRequest(request).mergeContext(customExtraContext).identify(id, trait);
```

Comment:
update copy

Reply:
I don't understand what should be fixed here.

Reply:
not sure what I meant here 🙈